packages/react-ui-core/src/Gmap/withGoogleScript.js

```diff
@@ -0,0 +1,92 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import autobind from 'autobind-decorator'
+import scriptjs from 'scriptjs'
+import { getDisplayName } from 'recompose'
+import { parseArgs } from '@rentpath/react-ui-utils'
+
+const API_BASE_URL = 'https://maps.googleapis.com/maps/api/js'
+
+export default function(BaseComponent) {
+  const factory = React.createFactory(BaseComponent)
```

Comment:
according to docs this method is discouraged

would something like this work? 

```
const withGoogleScript = Wrapped => wrappedProps => {
  <Wrapped ...wrappedProps><Container /></Wrapped>
}
```

Reply:
It's discouraged as it's legacy. `React.createElement` should be used instead which is all jsx is under the hood anyway.

Reply:
I'm not sure this is better:
```